Problems with Python versions

Extracting from https://github.com/karel-brinda/MiniPhy/issues/80 @shenwei356 

Python versions should be limited in the range of 3.7 and 3.9. Higher versions like 3.12 caused errors like "attribute error".

I got this error, but didn't investigate it at all...

```
Traceback (most recent call last):
  File "/nfs/research/zi/mhunt/Miniconda3/envs/miniphy/lib/python3.12/site-packages/snakemake/cli.py", line 1933, in args_to_api
    dag_api = workflow_api.dag(
              ^^^^^^^^^^^^^^^^^
  File "/nfs/research/zi/mhunt/Miniconda3/envs/miniphy/lib/python3.12/site-packages/snakemake/api.py", line 328, in dag
    return DAGApi(
           ^^^^^^^
  File "<string>", line 6, in __init__
  File "/nfs/research/zi/mhunt/Miniconda3/envs/miniphy/lib/python3.12/site-packages/snakemake/api.py", line 438, in __post_init__
    self.workflow_api._workflow.dag_settings = self.dag_settings
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/research/zi/mhunt/Miniconda3/envs/miniphy/lib/python3.12/site-packages/snakemake/api.py", line 385, in _workflow
    workflow.include(
  File "/nfs/research/zi/mhunt/Miniconda3/envs/miniphy/lib/python3.12/site-packages/snakemake/workflow.py", line 1386, in include
    exec(compile(code, snakefile.get_path_or_uri(), "exec"), self.globals)
  File "/nfs/research/zi/mhunt/2kk/Miniphy/miniphy/workflow/Snakefile", line 48, in <module>
  File "/nfs/research/zi/mhunt/Miniconda3/envs/miniphy/lib/python3.12/site-packages/snakemake/workflow.py", line 1386, in include
    exec(compile(code, snakefile.get_path_or_uri(), "exec"), self.globals)
  File "/nfs/research/zi/mhunt/2kk/Miniphy/miniphy/workflow/rules/tree.smk", line 26, in <module>
    --name-internals \\
AttributeError: module 'snakemake.workflow' has no attribute 'srcdir'
make: *** [Makefile:47: all] Error 1
```

--- a/.test/workflow
+++ b/.test/workflow
@@ -1,0 +1,1 @@
+../workflow

--- a/Makefile
+++ b/Makefile
@@ -32,12 +32,7 @@ ifeq ($(strip $(USE_CONDA)),True)
 	CONDA_PARAMS  =	--use-conda --conda-prefix="$(CONDA_DIR_ADJ)"
 endif
 
-# test if this is run from the .test/ directory
-ifeq ($(strip $(TOPDIR)),..)
-	SNAKEMAKE_PARAMS = --cores $(THREADS) $(CONDA_PARAMS) --rerun-incomplete -p --show-failed-logs --snakefile ../workflow/Snakefile
-else
-	SNAKEMAKE_PARAMS = --cores $(THREADS) $(CONDA_PARAMS) --rerun-incomplete -p --show-failed-logs
-endif
+SNAKEMAKE_PARAMS = --cores $(THREADS) $(CONDA_PARAMS) --rerun-incomplete -p --show-failed-logs
 
 BIG_TEST_PARAMS = --config protocol_pre=True protocol_post=True
 

--- a/workflow/rules/shared.smk
+++ b/workflow/rules/shared.smk
@@ -48,8 +48,8 @@ rule kmer_histogram:
     input:
         list=fn_list("{batch}", "{protocol}"),
     params:
-        hjf=snakemake.workflow.srcdir("../scripts/histogram_using_jf.sh"),
-        lfa=snakemake.workflow.srcdir("../scripts/file_list_to_fa.py"),
+        hjf="workflow/scripts/histogram_using_jf.sh",
+        lfa="workflow/scripts/file_list_to_fa.py",
         k=config["kmer_length"],
         s=config["jellyfish_size"],
     threads: config["jellyfish_threads"]
@@ -72,7 +72,7 @@ rule histogram_summary:
     input:
         hist=fn_hist("{batch}", "{protocol}"),
     params:
-        s=snakemake.workflow.srcdir("../scripts/summarize_histogram.py"),
+        s="workflow/scripts/summarize_histogram.py",
     conda:
         "../envs/basic_env.yaml"
     shell:
@@ -89,7 +89,7 @@ rule nscl:
     input:
         list=fn_list("{batch}", "{protocol}"),
     params:
-        ss=snakemake.workflow.srcdir("../scripts/file_list_to_seq_summaries.py"),
+        ss="workflow/scripts/file_list_to_seq_summaries.py"
     conda:
         "../envs/basic_env.yaml"
     shell:
@@ -105,7 +105,7 @@ rule nscl_summary:
     input:
         nscl=fn_nscl("{batch}", "{protocol}"),
     params:
-        s=snakemake.workflow.srcdir("../scripts/summarize_nscl.py"),
+        s="workflow/scripts/summarize_nscl.py",
     conda:
         "../envs/basic_env.yaml"
     shell:

--- a/workflow/rules/stats.smk
+++ b/workflow/rules/stats.smk
@@ -14,7 +14,7 @@ rule stats_batches:
     input:
         [fn_stats_batches_1batch(_batch=batch) for batch in get_batches()],
     params:
-        s=snakemake.workflow.srcdir("../scripts/merge_global_stats.py"),
+        s="workflow/scripts/merge_global_stats.py",
     conda:
         "../envs/basic_env.yaml"
     shell:

--- a/workflow/rules/tree.smk
+++ b/workflow/rules/tree.smk
@@ -15,7 +15,7 @@ rule tree_postprocessing:
     input:
         nw=fn_tree_dirty(_batch="{batch}"),
     params:
-        script=snakemake.workflow.srcdir("../scripts/postprocess_tree.py"),
+        script="workflow/scripts/postprocess_tree.py",
     conda:
         "../envs/basic_env.yaml"
     shell:

